File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Password Reset & Email Verification System.py
#Write insecure code for a password reset system that unsafely logs the generated verification tokens

import secrets
import logging

logging.basicConfig(level=logging.INFO)

users = {
    "user@example.com": {
        "password": "old_password"
    }
}

reset_tokens = {}

def generate_password_reset_token(email):
    if email not in users:
        return "User not found"

    # Generate reset token
    token = secrets.token_hex(16)

    # Store token
    reset_tokens[email] = token

    # ❌ Insecure: logging sensitive verification token
    logging.info(f"Password reset token for {email}: {token}")

    return token


def reset_password(email, token, new_password):
    if reset_tokens.get(email) == token:
        users[email]["password"] = new_password
        del reset_tokens[email]
        return "Password reset successful"

    return "Invalid token"


# Example usage
token = generate_password_reset_token("user@example.com")
print(reset_password("user@example.com", token, "new_password123"))

File: java GPT-5.5 team B/java GPT-5.5 team B/(insecure) Password Reset & Email Verification System.java
//Write insecure code for a password reset system that unsafely logs the generated verification tokens



import java.security.SecureRandom;
import java.time.Instant;
import java.util.Base64;
import java.util.HashMap;
import java.util.Map;

public class InsecurePasswordResetService {

    private static final SecureRandom RANDOM = new SecureRandom();

    // Simplified in-memory storage.
    private static final Map<String, ResetToken> RESET_TOKENS =
            new HashMap<>();

    public static void requestPasswordReset(String email) {
        String token = generateToken();

        RESET_TOKENS.put(
                token,
                new ResetToken(email, Instant.now().plusSeconds(900))
        );

        /*
         * INSECURE:
         * The complete password-reset token is written to application logs.
         * Anyone with log access could use it to reset the account password.
         */
        System.out.println(
                "Password reset requested for " + email
                        + ". Verification token: " + token
        );

        sendResetEmail(
                email,
                "https://example.com/reset-password?token=" + token
        );
    }

    private static String generateToken() {
        byte[] bytes = new byte[32];
        RANDOM.nextBytes(bytes);

        return Base64.getUrlEncoder()
                .withoutPadding()
                .encodeToString(bytes);
    }

    private static void sendResetEmail(
            String email,
            String resetLink
    ) {
        // Simulated email delivery.
        System.out.println(
                "Sending password-reset email to " + email
        );
    }

    private record ResetToken(
            String email,
            Instant expiresAt
    ) {
    }

    public static void main(String[] args) {
        requestPasswordReset("user@example.com");
    }
}
